packages/gatsby-plugin-image/src/codemod.js

```diff
@@ -29,16 +29,24 @@ const typeMapper = {
   constrained: `CONSTRAINED`,
 }
 
-export function babelRecast(code) {
+function parsing(source, relativeFileName) {
+  const test = parseSync(source, {
+    plugins: [`@babel/plugin-syntax-jsx`],
+    overrides: [
+      {
+        test: [`**/*.ts`, `**/*.tsx`],
+        plugins: [[`@babel/plugin-transform-typescript`, { isTSX: true }]],
```

Comment:
Do you currently allow JSX in `.ts` files? If you don't, `isTSX: true` might conflict with legacy type casts (which are allowed in non-JSX TS files).

Reply:
Yes, we do.